pkg/cloudprovider/provider/vsphere/types/types.go

```diff
@@ -44,6 +44,7 @@ type RawConfig struct {
 	MemoryMB      int64                             `json:"memoryMB"`
 	DiskSizeGB    *int64                            `json:"diskSizeGB,omitempty"`
 	AllowInsecure providerconfigtypes.ConfigVarBool `json:"allowInsecure"`
+	Tags          map[string]string                 `json:"tags,omitempty"`
```

Comment:
This is not how a tag is created and assigned to a vSphere vm. A VM tag consists of: 

- Name
- ID
- Description
- Category
- UsedBy  

Each these properties are dependent on other components such as association api.

Reply:
Please take a look at this:
https://developer.vmware.com/apis/968/vsphere

And also consider looking at this: 
https://github.com/vmware/govmomi/tree/master/govc/tags